src/devices/Nmea0183/Sentences/RecommendedMinimumNavigationInformation.cs

```diff
@@ -0,0 +1,300 @@
+            } 
+            else if (time.Length != 0)
+            {
+                dateTime = DateTimeOffset.ParseExact(time, "HHmmss.ff", formatProvider: null);
+            }
+            else if (date.Length != 0)
+            {
+                dateTime = DateTimeOffset.ParseExact(date, "ddMMyy", formatProvider: null);
+            }
+
+            double? mag = ReadValue();
+            CardinalDirection? magTurn = (CardinalDirection?)ReadChar();
+
+            // handle undocumented field
+            // per spec we should not have any extra fields but NEO-M8 does have them
+            if (field.MoveNext())
+            {
+                string val = field.Current;
+                Status2 = string.IsNullOrEmpty(val) ? (NavigationStatus?)null : (NavigationStatus?)val.Single();
+            }
+
+            Done();
+
+            DateTime = dateTime;
+            Status = status;
+            _latitude = lat;
+            _latitudeTurn = latTurn;
+            _longitude = lon;
+            _longitudeTurn = lonTurn;
+            SpeedOverGroundInKnots = speed;
+            TrackMadeGoodInDegreesTrue = track;
+            _positiveMagneticVariationInDegrees = mag;
+            _magneticVariationTurn = magTurn;
+        }
+
+        public RecommendedMinimumNavigationInformation(
+            DateTimeOffset? dateTime,
+            NavigationStatus? status,
+            double? latitude,
+            double? longitude,
+            double? speedOverGroundInKnots,
+            double? trackMadeGoodInDegreesTrue,
+            double? magneticVariationInDegrees)
+        {
+            DateTime = dateTime;
+            Status = status;
+            LatitudeDegrees = latitude;
+            LongitudeDegrees = longitude;
+            SpeedOverGroundInKnots = speedOverGroundInKnots;
+            TrackMadeGoodInDegreesTrue = trackMadeGoodInDegreesTrue;
+            MagneticVariationInDegrees = magneticVariationInDegrees;
+        }
+
+        private static int DirectionToSign(CardinalDirection direction)
+        {
+            switch (direction)
+            {
+                case CardinalDirection.North:
+                case CardinalDirection.East:
+                    return 1;
+                case CardinalDirection.South:
+                case CardinalDirection.West:
+                    return -1;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(direction));
+            }
+        }
+
+        private static double? Nmea0183ToDegrees(double? degreesMinutes, CardinalDirection? direction)
+        {
+            if (!degreesMinutes.HasValue || !direction.HasValue)
+                return null;
+
+            // ddddmm.mm
+            double degrees = Math.Floor(degreesMinutes.Value / 100);
+            double minutes = degreesMinutes.Value - (degrees * 100);
+            return ((double)degrees + (double)minutes / 60.0) * DirectionToSign(direction.Value);
```

Comment:
Why is it required to cast degrees and minutes (which have `double` type) to `double`?